ssl/statem/statem_srvr.c

```diff
@@ -528,14 +528,13 @@ OCSP_RESPONSE *get_ocsp_response(SSL_CONNECTION *s, int chainidx)
                 /* use the first single response to get the algorithm used */
                 cid = (OCSP_CERTID *)OCSP_SINGLERESP_get0_id(sr);
 
-                EVP_set_default_properties(sctx->libctx, sctx->propq);
-
                 /* determine the md algorithm which was used to create cert id */
                 OCSP_id_get0_info(&respIssuerNameHash, &cert_id_md_oid, NULL, &respSerial, cid);
                 if (cert_id_md_oid != NULL)
-                    cert_id_md = EVP_get_digestbyobj(cert_id_md_oid);
+                    cert_id_md = EVP_MD_fetch(sctx->libctx,
+                                              OBJ_nid2sn(OBJ_obj2nid(cert_id_md_oid)), sctx->propq);
```

Comment:
I think using OBJ_obj2txt() would be better way as it avoids going through nids. Please note that algorithms should be fetchable by the textual representation of the oid.